src/coreclr/vm/jitinterface.cpp

```diff
@@ -8868,8 +8868,40 @@ CORINFO_METHOD_HANDLE CEEInfo::getUnboxedEntry(
         MethodTable* pMT = pMD->GetMethodTable();
         MethodDesc* pUnboxedMD = pMT->GetUnboxedEntryPointMD(pMD);
 
-        result = (CORINFO_METHOD_HANDLE)pUnboxedMD;
-        requiresInstMTArg = !!pUnboxedMD->RequiresInstMethodTableArg();
+        // Give up on methods which may potentially expose byrefs to struct's fields:
+        // * byref return types
+        // * byref-like return types
+        // * byref to byref-like types in parameters
```

Comment:
What about a native pointer to a byref like type? Or worse, a void pointer cast in the body to a pointer to a byref like type.

I'm curious what exposed the problem here. It reminds me of the "unboxing of nullables" case a little bit where we don't produce a heap pointer. I wonder if we have to spec ourselves into being allowed to do this optimization.